org.eclipse.draw2d/src/org/eclipse/draw2d/SWTGraphics.java

```diff
@@ -335,7 +335,10 @@ private void checkSharedClipping() {
 			sharedClipping = false;
 
 			boolean previouslyApplied = (appliedState == currentState.relativeClip);
-			currentState.relativeClip = currentState.relativeClip.getCopy();
+			// Fix: currentState.relativeClip can be null and lead to NPE
+			currentState.relativeClip = currentState.relativeClip != null
```

Comment:
wouldn't a simple if be nicer here? setting it to null when it is already null seems for me a bit confusing.